### ISSUE TITLE
code is broken, cache contained real data

### ISSUE THREAD

**Issue by gongshichina** [etienne87/pytorch-stream-dataloader#8]
<img width="797" alt="2b2ad71bd69a5b32b1c2aaeedad0a5a0" src="https://user-images.githubusercontent.com/42108203/217557616-6988e3da-bef1-4386-b024-37c219a7b49d.png">
 
Thanks for your good job!
 
I am wondering why the problem '''code is broken' occurs. I have tried to get the num worker from
 'torch.utils.data.get_worker_info()', but it still throws error.

![image](https://user-images.githubusercontent.com/42108203/217558926-826bd91b-9870-4a72-bdba-33019fb24fdf.png)

**Comment by etienne87** [comment]
Thanks a lot for reporting the bug! this is a real bug showing an error in design/ implementation from me, because after exhausting real data from stream dataset workers, somehow the queue still contain some real data (not padding data). Could you provide some minimalistic StreamDataset to reproduce the bug?  In the mean time you can just comment this part if you just do not want any crash.
